src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -80,6 +83,9 @@ export class TerminalService implements ITerminalService {
 			this.setActiveInstanceByIndex(0);
 		}
 		this._onInstancesChanged.fire();
+		this._updateCursorBlink(terminalInstance);
```

Comment:
These calls would be more at home inside `TerminalInstance.constructor`. Thinking about this further we may be able to get rid of both calls in `TerminalPanel.setVisible` as well by updating the config at the end of `TerminalInstance.attachToElement`. I believe that covers the cases we need to worry about for updating the config:

- When the terminal is created and the panel exists (`attachToElement` will trigger immediately)
- When the terminal is created via the API and the panel does not exist (`attachToElement` call will be deferred)
- When the config is updated (`TerminalService` now listens to `IConfigurationservice.onDidUpdateConfiguration`)